src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java

```diff
@@ -139,56 +169,133 @@ public Integer getActiveUsers(Long span) {
+     * @param endDate the endDate
+     * @return an array, containing the values for each bar in the graph
      */
-    private Integer[] createSubmissionCountArrayForYear(List<Map<String, Object>> outcome, Integer[] result, ZonedDateTime currentDate) {
+    private Integer[] createResultArrayForYear(List<Map<String, Object>> outcome, Integer[] result, ZonedDateTime endDate) {
         for (Map<String, Object> map : outcome) {
             ZonedDateTime date = (ZonedDateTime) map.get("day");
             Integer amount = map.get("amount") != null ? ((Long) map.get("amount")).intValue() : null;
             for (int i = 0; i < 12; i++) {
-                if (date.getMonth() == currentDate.minusMonths(i).getMonth() && date.getYear() == currentDate.minusMonths(i).getYear()) {
+                if (date.getMonth() == endDate.minusMonths(i).getMonth()) {
                     result[11 - i] += amount;
                 }
             }
         }
         return result;
     }
 
-    public Integer getReleasedExercises(Long span) {
-        return this.statisticsRepository.getReleasedExercises(ZonedDateTime.now().minusDays(span), ZonedDateTime.now());
-    }
-
-    public Integer getExerciseDeadlines(Long span) {
-        return this.statisticsRepository.getExerciseDeadlines(ZonedDateTime.now().minusDays(span), ZonedDateTime.now());
-    }
-
-    public Integer getConductedExams(Long span) {
-        return this.statisticsRepository.getConductedExams(ZonedDateTime.now().minusDays(span), ZonedDateTime.now());
-    }
-
-    public Integer getExamParticipations(Long span) {
-        return this.statisticsRepository.getExamParticipations(ZonedDateTime.now().minusDays(span));
-    }
-
-    public Integer getExamRegistrations(Long span) {
-        return this.statisticsRepository.getExamRegistrations(ZonedDateTime.now().minusDays(span), ZonedDateTime.now());
-    }
-
-    public Integer getActiveTutors(Long span) {
-        return this.statisticsRepository.getActiveTutors(ZonedDateTime.now().minusDays(span));
-    }
-
-    public Integer getCreatedResults(Long span) {
-        return this.statisticsRepository.getCreatedResults(ZonedDateTime.now().minusDays(span));
+    /**
+    * This method handles the duplicity of usernames in the active user call. It gets a List<Map<String, Object>> analogue to previous methods, but instead of numbers in an amount key,
+    * it contains a username key with the actual username as value. It then handles all the usernames and returns a List<Map<String, Object>>, but now with the the key "amount"
+    * and value the number of users in this interval
+    *
+    * @param span DAY,WEEK,MONTH or YEAR
+    * @param result the result given by the Repository call
+    * @param startDate the startDate of the period
+    * @return A List<Map<String, Object>> analogue to other database calls
+    */
+    private List<Map<String, Object>> convertMapList(SpanType span, List<Map<String, Object>> result, ZonedDateTime startDate) {
```

Comment:
This method really looks like it has some duplication. Can we do something like the following?
```
private List<Map<String, Object>> convertMapList(SpanType span, List<Map<String, Object>> result, ZonedDateTime startDate) {
	List<Map<String, Object>> returnList = new ArrayList<>();
	Map<Integer, List<String>> users = new HashMap<>();
	
	for (Map<String, Object> listElement : result) {
		Object index;
		ZonedDateTime date = (ZonedDateTime) listElement.get("day");
		if (span == DAY) {
			index = date.getHour();
		} else if (span == WEEK || span == MONTH) {
			index = date.getDayOfMonth();
		} else {
			index = date.getMonth();
		}

		String username = listElement.get("username").toString();
		List<String> usersInSameSlot = users.get(index);
		if (usersInSameSlot == null) {
			usersInSameSlot = new ArrayList<>();
			usersInSameSlot.add(username);
			users.put(index, usersInSameSlot);
		} else if (!usersInSameSlot.contains(username))) {
			usersInSameSlot.add(username);
			// btw: I don't think we need to put this again here, since we're modifying the arraylist by reference?
			// users.put(index, usersInSameSlot);
		}
	}
	users.forEach((k, v) -> {
		Object start;
		if (span == DAY) {
			start = startDate.withHour(k);
		} else if (span == WEEK || span == MONTH) {
			start = startDate.withDayOfMonth(k);
		} else {
			start = startDate.withMonth(getMonthIndex(k));
		}
	
		Map<String, Object> listElement = new HashMap<>();
		listElement.put("day", start);
		listElement.put("amount", (long) v.size());
		returnList.add(listElement);
	});
	
	return returnList;
}
```

Reply:
Very nice idea with the index as Object, did not think of that! I adapted your suggestion a bit and implemented it [here](https://github.com/ls1intum/Artemis/pull/2511/commits/0e63a45bd195a1e39a220f42840ea55ed66660b0)